core/domain/story_domain.py

```diff
@@ -320,6 +324,188 @@ def update_story_contents_from_model(
         versioned_story_contents['story_contents'] = conversion_fn(
             versioned_story_contents['story_contents'])
 
+    def update_title(self, title):
+        """Updates the title of the story.
+
+        Args:
+            title: str. The new title of the story.
+        """
+        self.title = title
+
+    def update_description(self, description):
+        """Updates the description of the story.
+
+        Args:
+            description: str. The new description of the story.
+        """
+        self.description = description
+
+    def update_notes(self, notes):
+        """Updates the notes of the story.
+
+        Args:
+            notes: str. The new notes of the story.
+        """
+        self.notes = notes
+
+    def update_language_code(self, language_code):
+        """Updates the language code of the story.
+
+        Args:
+            language_code: str. The new language code of the story.
+        """
+        self.language_code = language_code
+
+    def add_node(self, node_id):
+        """Adds a new default node with the given node_id.
+
+        Args:
+            node_id: str. The id of the node.
+        """
+        self.story_contents.nodes.append(
+            StoryNode.create_default_story_node(node_id))
+
+    def _find_node(self, node_id):
+        """Returns the index of the story node with the given node
+        id, or None if the node id is not in the story contents dict.
+
+        Args:
+            node_id: str. The id of the node.
+
+        Returns:
+            int or None. The index of the corresponding node, or None if there
+            is no such node.
+        """
+        for ind, node in enumerate(self.story_contents.nodes):
+            if node.id == node_id:
+                return ind
+        return None
+
+    def _check_exploration_id_already_present(self, exploration_id):
+        """Returns whether a node with the given exploration id is already
+        present in story_contents.
+
+        Args:
+            exploration_id: str. The id of the exploration.
+
+        Returns:
+            Boolean. True if a node with the given exploration id is already
```

Comment:
bool. Whether a node with the given exploration ID is already present.

Reply:
Done.